napari/utils/naming.py

```diff
@@ -61,16 +64,25 @@ def magic_name(value, *, path_prefix):
     name : str or None
         Name of the variable, if found.
     """
-    frame = inspect.currentframe().f_back
-    code = frame.f_code
-
-    while code.co_filename.startswith(path_prefix):
-        frame = frame.f_back
+    try:
+        frame = inspect.currentframe().f_back
```

Comment:
@mkitti thanks for this!

One thing I would say is that generally you want to wrap the smallest amount of code possible in a try/except, so you are not accidentally catching things you didn't mean to catch.

In this case, the first `currentframe().f_back` is always guaranteed to be within napari and therefore exist (?) (my lack of knowledge of the Julia/Python interface is probably showing :sweat_smile:). In that case, you could wrap just one line in try/except: `frame = frame.f_back`.

Even if not, I'd suggest wrapping both `f_back` lines in try/except and leaving the other lines untouched.

How does this sound?

Also, can you think of a simple way that this can be tested? It's fine if not, just thought it would be worth thinking about.

Reply:
Let's itemize where the potential issues are first:
1. `inspect.currentframe()` could be `None` if the underlying Python implementation lacks stack frame support causing an `AttributeError` on master ( 5e8be09 ) Line 64.
https://docs.python.org/3/library/inspect.html#inspect.currentframe
2. `frame = currentframe().f_back` could be `None` depending on the caller within Python on Line 64 causing an error on Line 65 when trying to obtain `frame.f_code`
3. `frame.f_back` could be `None` on Line 72 depending on the caller within Python resulting in `AttributeError` on line Line 73.

Additonally, it might make sense to keep current structure and add a `finally` to delete `frame` explicitly to avoid a stack frame leak:
https://docs.python.org/3/library/inspect.html#the-interpreter-stack
https://bugs.python.org/issue543148

In terms of control flow, the current situation makes sense to me in a structured programming sense. If an exception occurs anywhere in the try block, we should stop executing the rest of the try block. 

In light of this, I have a more explicit refactoring in mind if we want to check each line.